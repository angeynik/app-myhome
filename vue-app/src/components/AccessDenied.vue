<template>
    <div>
      <h1>403 - Доступ запрещен</h1>
      <p>У вас недостаточно прав для просмотра этой страницы.</p>
      <router-link to="/login">Авторизация</router-link>
      <router-link to="/">Вернуться на главную</router-link>
    </div>
  </template>
  
  <script>
  export default {
    name: 'AccessDenied',
  };
  </script>
  
  <style scoped>
  /* Стили для страницы ошибки */
  </style>