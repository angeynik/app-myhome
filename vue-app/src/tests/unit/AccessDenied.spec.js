import { mount } from '@vue/test-utils';
import AccessDenied from '@/components/AccessDenied.vue';
import { createStore } from 'vuex';

describe('AccessDenied.vue', () => {
  let store;
  
  beforeEach(() => {
    store = createStore({
      state: {},
      getters: {
        level: () => 2,
        isAuthenticated: () => true
      }
    });
  });

  it('matches the snapshot', () => {
    const wrapper = mount(AccessDenied, {
      global: {
        plugins: [store]
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});