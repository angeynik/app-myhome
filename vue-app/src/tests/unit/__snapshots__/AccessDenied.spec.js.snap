// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessDenied.vue matches the snapshot 1`] = `
"<div class="header-bottom">
  <nav>
    <router-link-stub to="/login"></router-link-stub> | <router-link-stub to="/dashboard"></router-link-stub> | <router-link-stub to="/smart-home"></router-link-stub> | <router-link-stub to="/manufact-automatation"></router-link-stub> |
    <!-- Profile (доступна для level 2 и 3) -->
    <!--v-if-->
    <!-- Users (доступна для level 3) -->
    <!--v-if-->
  </nav>
</div>
<p style="width: 100%; height: 1px;"></p>
<div>
  <h1>403 - Доступ запрещен</h1>
  <p>У вас недостаточно прав для просмотра этой страницы.</p>
  <router-link-stub to="/login"></router-link-stub>
  <router-link-stub to="/"></router-link-stub>
</div>"
`;
